src/settings_gui.cpp

```diff
@@ -759,32 +887,34 @@ struct GameOptionsWindow : Window {
 				break;
 			}
 
-#ifdef HAS_TRUETYPE_FONT
-			case WID_GO_GUI_FONT_SPRITE:
+			case WID_GO_FONT_SPRITE:
 				_fcsettings.prefer_sprite = !_fcsettings.prefer_sprite;
 
-				this->SetWidgetLoweredState(WID_GO_GUI_FONT_SPRITE, _fcsettings.prefer_sprite);
-				this->SetWidgetDisabledState(WID_GO_GUI_FONT_AA, _fcsettings.prefer_sprite);
+				this->SetWidgetLoweredState(WID_GO_FONT_SPRITE, _fcsettings.prefer_sprite);
+				this->SetWidgetDisabledState(WID_GO_FONT_AA, _fcsettings.prefer_sprite);
+				this->SetWidgetDisabledState(WID_GO_FONT_SMALL_FONT_SIZE_SLIDER, _fcsettings.prefer_sprite && GetFontCacheSubSetting(FS_SMALL)->font.empty());
```

Comment:
Question: Why does the font need to be empty to disable this? Wouldn't the "Prefer sprite font" be enough?

Reply:
Earlier I made the mistake of assuming that the  prefer sprites checkbox should override all current font settings.  I think it was Peter that corrected me either in chat or in one of these reviews.

The prefer sprites checkbox should only affect the default OpenTTD fonts but, not any other fonts that the user has manually set. Checking for empty font used to be a good way to check for default fonts.  There's a new function for that I could use instead now. I'll update it to call that function.